cleanlab/multiannotator.py

```diff
@@ -0,0 +1,436 @@
+import numpy as np
+import pandas as pd
+from typing import List, Union, Tuple
+from cleanlab.rank import get_label_quality_scores
+from cleanlab.dataset import overall_label_health_score, _get_worst_class
+import warnings
+
+# TODO: unsure where to place this function? should it go to cleanlab.rank or stay here
+def _get_label_quality_scores_with_NA(
+    labels: pd.Series,
+    pred_probs: np.array,
+    kwargs: dict = {},
+) -> pd.Series:
+    label_quality_scores_subset = get_label_quality_scores(
+        labels=labels[pd.notna(labels)],
+        pred_probs=pred_probs[pd.notna(labels)],
+        **kwargs,
+    )
+    label_quality_scores = pd.Series(np.nan, index=np.arange(len(labels)))
+    label_quality_scores[pd.notna(labels)] = label_quality_scores_subset
+    return label_quality_scores
+
+
+def get_consensus_labels(
+    labels_multiannotator: pd.DataFrame,
+    pred_probs: np.array,
+    method: str,
+) -> np.array:
+    """Returns consensus labels aggregated from multiple annotators, determined by method specified.
+
+    Parameters
+    ----------
+    labels_multiannotator : pd.DataFrame
+        2D pandas DataFrame of multiple given labels for each example with shape (N, M),
+        where N is the number of examples and M is the number of annotators.
+        For more details, labels in the same format expected by the :py:func:`get_label_quality_scores_multiannotator <cleanlab.multiannotator.get_label_quality_scores_multiannotator>`.
+
+    pred_probs : np.array
+        An array of shape ``(N, K)`` of model-predicted probabilities, ``P(label=k|x)``.
+        For details, predicted probabilities in the same format expected by the :py:func:`get_label_quality_scores_multiannotator <cleanlab.multiannotator.get_label_quality_scores_multiannotator>`.
+
+    method : str (Options: ["majority"])
+        The method used to aggregate labels from multiple annotators into a single consensus label. Default is by using the majority vote.
+        Options include:
+        - ``"majority"``: means consensus labels are reached via a simple majority vote among annotators, with ties broken via ``pred_probs``.
+
+    Returns
+    -------
+    consensus_labels: np.array
+        An array of shape ``(N,)`` with the consensus labels aggregated from all annotators.
+    """
+
+    valid_methods = ["majority"]
+    consensus_labels = np.empty(len(labels_multiannotator), dtype="int32")
+
+    # TODO: is there a more efficient/elegant way to do this?
+    if method == "majority":
+        mode_labels_multiannotator = labels_multiannotator.mode(axis=1)
+        for i in range(len(mode_labels_multiannotator)):
```

Comment:
Can think more about this, but at least you should only have to loop over those examples `i` where `mode_labels_multiannotator[i]` is longer than 1 in the first place right?  So restricting the loop to only those examples should already provide big speedup (could even have special handling of examples with only 2 labels if you've already computed `num_annotations` at this point, since we know these will have ties).

Reply:
Another alternative is to do `value_counts()` and implement the mode + tie-breaking yourself (eg. could simply add the pred_probs to the value_counts if called with `sort=False`). 

But would probably first want to some timing benchmarks/googling to see how much slower `value_counts()` is in comparison since it involves extra computation than `mode()`.

Reply:
Updated so that less computation is done if `len(mode_labels_multiannotator[i]) == 1`, but still requires looping over all examples